src/Microsoft.AspNetCore.SignalR.Common/Internal/Protocol/Utf8BufferTextReader.cs

```diff
@@ -11,47 +11,64 @@ namespace Microsoft.AspNetCore.SignalR.Internal.Protocol
     internal class Utf8BufferTextReader : TextReader
     {
         private ReadOnlyMemory<byte> _utf8Buffer;
+        private Decoder _decoder;
 
-        public Utf8BufferTextReader(ReadOnlyMemory<byte> utf8Buffer)
+        [ThreadStatic]
+        private static Utf8BufferTextReader _cachedInstance;
+
+        public Utf8BufferTextReader()
         {
-            _utf8Buffer = utf8Buffer;
+            _decoder = Encoding.UTF8.GetDecoder();
         }
 
-        public override int Read(char[] buffer, int index, int count)
+        public static Utf8BufferTextReader Get(ReadOnlyMemory<byte> utf8Buffer)
         {
-            if (_utf8Buffer.IsEmpty)
+            var reader = _cachedInstance;
+            if (reader == null)
```

Comment:
When reader is not null I want a flag set on it to say that it is in use, and the flag is reset on `Return`. If `Get` is called and the cached instance is already in use then throw a friendly error message.

It probably won't happen but better to be safe the sorry.

Reply:
@JamesNK I got you covered bro 😄 .